src/sagemaker_xgboost_container/algorithm_mode/serve.py

```diff
@@ -225,19 +226,19 @@ def invocations():
         logging.exception(e)
         return flask.Response(response="Unable to evaluate payload provided: %s" % e, status=http.client.BAD_REQUEST)
 
-    if serve_utils.is_selectable_inference_output():
-        try:
-            accept = _parse_accept(flask.request)
-        except Exception as e:
-            logging.exception(e)
-            return flask.Response(response=str(e), status=http.client.NOT_ACCEPTABLE)
+    try:
+        accept = _parse_accept(flask.request)
+    except Exception as e:
+        logging.exception(e)
+        return flask.Response(response=str(e), status=http.client.NOT_ACCEPTABLE)
 
+    if serve_utils.is_selectable_inference_output():
         return _handle_selectable_inference_response(preds, accept)
 
     if SAGEMAKER_BATCH:
         return_data = "\n".join(map(str, preds.tolist())) + '\n'
     else:
-        return_data = ",".join(map(str, preds.tolist()))
+        return_data = encoders.encode(preds.tolist(), accept)
 
     return flask.Response(response=return_data, status=http.client.OK, mimetype="text/csv")
```

Comment:
Does this `mimetype` need to be updated too?

Reply:
good catch, updating on next commit